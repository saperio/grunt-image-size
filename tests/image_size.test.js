/* eslint-env jest */

const { runGrunt, grunt } = require('./utils/grunt')
const { config, file: { expand, readJSON } } = grunt

const buildPath = config('path.build')

describe('Image Size task', () => {
  beforeEach(() => runGrunt(['clean']))

  it('should produce correct data file with defaults for single source', () =>
    runGrunt(['image_size:singleWithDefaults']).then(() =>
      expect(readJSON(buildPath.singleWithDefaults)).toMatchSnapshot()
    )
  )

  it('should produce correct data file with defaults for multiple sources', () =>
    runGrunt(['image_size:multipleWithDefaults']).then(() =>
      expect(readJSON(buildPath.multipleWithDefaults)).toMatchSnapshot()
    )
  )

  it('should produce correct data file with defaults for multiple nested sources', () =>
    runGrunt(['image_size:nestedWithDefaults']).then(() =>
      expect(readJSON(buildPath.nestedWithDefaults)).toMatchSnapshot()
    )
  )

  it('should produce correct data files with defaults for multiple nested expanded sources', () =>
    runGrunt(['image_size:expandedNestedWithDefaults']).then(() => {
      const buildedFilesPaths = expand(
        { filter: 'isFile' },
        `${buildPath.expandedNestedWithDefaults}/**`
      )

      expect(buildedFilesPaths).toMatchSnapshot()

      buildedFilesPaths.forEach((filepath) =>
        expect(readJSON(filepath)).toMatchSnapshot()
      )
    })
  )

  // it('should error on file with error', () =>
  //   expect(runGrunt(['image_size:withError'])).rejects.toBeDefined()
  // )
})
