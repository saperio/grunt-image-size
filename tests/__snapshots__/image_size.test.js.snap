// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image Size task should produce correct data file with defaults for multiple nested sources 1`] = `
Array [
  Object {
    "height": 150,
    "name": "tests/fixtures/nested/image-1.jpg",
    "width": 100,
  },
  Object {
    "height": 250,
    "name": "tests/fixtures/nested/image-2.png",
    "width": 200,
  },
  Object {
    "height": 350,
    "name": "tests/fixtures/nested/level-2/image-3.svg",
    "width": 300,
  },
]
`;

exports[`Image Size task should produce correct data file with defaults for multiple sources 1`] = `
Array [
  Object {
    "height": 150,
    "name": "tests/fixtures/multiple/image-1.jpg",
    "width": 100,
  },
  Object {
    "height": 250,
    "name": "tests/fixtures/multiple/image-2.png",
    "width": 200,
  },
  Object {
    "height": 350,
    "name": "tests/fixtures/multiple/image-3.svg",
    "width": 300,
  },
]
`;

exports[`Image Size task should produce correct data file with defaults for single source 1`] = `
Array [
  Object {
    "height": 150,
    "name": "tests/fixtures/single/image-1.jpg",
    "width": 100,
  },
]
`;

exports[`Image Size task should produce correct data files with defaults for multiple nested expanded sources 1`] = `
Array [
  "build/expandedNestedWithDefaults/image-1.json",
  "build/expandedNestedWithDefaults/image-2.json",
  "build/expandedNestedWithDefaults/level-2.json",
  "build/expandedNestedWithDefaults/level-2/image-3.json",
]
`;

exports[`Image Size task should produce correct data files with defaults for multiple nested expanded sources 2`] = `
Array [
  Object {
    "height": 150,
    "name": "tests/fixtures/nested/image-1.jpg",
    "width": 100,
  },
]
`;

exports[`Image Size task should produce correct data files with defaults for multiple nested expanded sources 3`] = `
Array [
  Object {
    "height": 250,
    "name": "tests/fixtures/nested/image-2.png",
    "width": 200,
  },
]
`;

exports[`Image Size task should produce correct data files with defaults for multiple nested expanded sources 4`] = `Array []`;

exports[`Image Size task should produce correct data files with defaults for multiple nested expanded sources 5`] = `
Array [
  Object {
    "height": 350,
    "name": "tests/fixtures/nested/level-2/image-3.svg",
    "width": 300,
  },
]
`;
